### PR TITLE
feat: Adds new configuration options to add custom tags (labels) to logs

### LINKED ISSUE #2716: Attach labels (tags) to logs
### Description

We need to support adding labels (tags) to logs.

TLDR:
- Labels disabled by default
- Labels should be prefixed with "tags."
- Labels are added to common attributes, not individual logs.
- Allows individual labels to be excluded

Dependent on this issue that adds configurations to the agents to enable this feature: https://github.com/newrelic/node-newrelic/issues/2713

### Implementation details:
- If `application_logging.forwarding.labels.enabled` is set to `true`, all labels **MUST** be attached in `common` attributes unless there is matching entry in `application_logging.forwarding.labels.exclude`.
- The attribute key **MUST** be the prefix `tags.` followed by the `label name`.
- Except for `tags.`, the attribute **MUST** match the case of the original label. Example, `MYLABEL;Value1` becomes `"tags.MYLABEL": "Value1"`.
- Add supportability metric for labels - `Supportability/Logging/Labels/{language}/{enabled\|disabled}`. **MUST** be generated by agents when labels on application logging is enabled. `disabled` **SHOULD** be generated if possible, although some agents' implementation may preclude it.  It is expected to be sent once per agent connect (not per harvest).|

Example newrelic.config bit from .NET showing labels with no exclude configuration:
```
<labels>label1:value1;LABEL2:value2;LABEL2-ALSO:value3</labels>
```

Example `log_event_data` payload based on above newrelic.config:
```
[{
   "common": {
     "attributes": {
       "entity.guid": <ENTITY_GUID>,
       "entity.name": "My Application",
       "hostname": "my-host-name",
       "tags.label1": "value1",
       "tags.LABEL2": "value2",
       "tags.LABEL2-ALSO":"value3"
     }
   },
   "logs": [{
       "timestamp": <TIMESTAMP_IN_UNIX_EPOCH_MILLSECONDS>,
       "level": "INFO"
       "message": "User 'xyz' logged in",
       "span.id": "123456789ADF",
       "trace.id": "ADF09876565"
     }]
}]
```

Same example above, but with the following exclude configuration (.NET agent XML):
```
<applicationLogging>
  <forwarding>
    <labels enabled="true" exclude="label2" />
  </forwarding>
</applicationLogging>
```

Example `log_event_data` payload based on above newrelic.config - no longer has LABEL2, but still has LABEL2-ALSO:
```
[{
   "common": {
     "attributes": {
       "entity.guid": <ENTITY_GUID>,
       "entity.name": "My Application",
       "hostname": "my-host-name",
       "tags.label1": "value1",
       "tags.LABEL2-ALSO":"value3"
     }
   },
   "logs": [{
       "timestamp": <TIMESTAMP_IN_UNIX_EPOCH_MILLSECONDS>,
       "level": "INFO"
       "message": "User 'xyz' logged in",
       "span.id": "123456789ADF",
       "trace.id": "ADF09876565"
     }]
}]
```
[Link to labels spec](https://source.datanerd.us/agents/agent-specs/blob/main/Labels.md#labels). 

[Discovery doc ](https://newrelic.atlassian.net/wiki/spaces/~712020468958465767472dae555831f9e3cc31/pages/3678240795/APM+Agent+Logging+Include+APM+Custom+Tags+in+Logs+For+Better+Insights)about labels.

### PR DESCRIPTION
**Pull request by svetlanabrennan**
## Description

- Added feature to support adding labels (tags) to logs. 
- Added unit tests

## How to Test

Run unit tests. 

## Related Issues

Closes #2716 

## TODO

- [x] Add supportability metrics

### CODE REVIEW

File: lib/aggregators/log-aggregator.js [hunk 1]
```diff
@@ -57,12 +59,32 @@ class LogAggregator extends EventAggregator {
...
+    const excludeSet = new Set(
+      this.agent.config.application_logging.forwarding.labels.exclude.map((item) =>
+        item.toLowerCase()
+      )
+    )
+
+    if (isLogLabelingEnabled(this.agent.config)) {
+      const configuredLabels = parseLabels(this.agent.config.labels)
+
+      configuredLabels.forEach((label) => {
```
[the remark this answers is not in the document]
Comment (bizob2828) [reply lib/aggregators/log-aggregator.js:80]: you can also prob do this in `lib/config/index.js` to reconcile what's the labels to keep as this is static

**Review by jsumners-nr — APPROVED** [review]
I'm confused as to why we apply labels to all logs instead of providing a way for users to add labels to specific logs, but the implementation looks sound to me.

File: test/unit/aggregators/log-aggregator.test.js [hunk 2]
```diff
@@ -175,6 +191,71 @@ test('Log Aggregator', async (t) => {
...
+
+  await t.test('should increment supportability metrics accordingly', (t) => {
+    const { agent } = t.nr
+    const expectedMetrics = [
+      [{ name: 'Supportability/Logging/Forwarding/Nodejs/enabled' }],
+      [{ name: 'Supportability/Logging/LocalDecorating/Nodejs/enabled' }],
+      [{ name: 'Supportability/Logging/Metrics/Nodejs/enabled' }],
+      [{ name: 'Supportability/Logging/Labels/Nodejs/enabled' }]
+    ]
+    assertMetrics(agent.metrics, expectedMetrics, false, false)
```
Comment (jsumners-nr) [inline test/unit/aggregators/log-aggregator.test.js:235]: I think what you're really going for here is:

```suggestion
    assertMetrics(agent.metrics, expectedMetrics, { exact: false })
```

**Review by bizob2828 — APPROVED** [review]
approving but since I made updates on @svetlanabrennan's behalf, I'll wait for @jsumners-nr to approve as well

**Review by jsumners-nr — APPROVED** [review]

**Pull request merged**